<p>The form $i++ creates a temporary variable whereas ++$i does not. It save CPU cycles.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$i++
</pre>
<h2>Compliant Solution</h2>
<pre>++$i</pre>
<h2>The three sources of impacts of a code identified are: </h2>
- Energy (measured in joules)
<br />    - Transfer (measured in Bytes)
<br />    - Storage (measured in Bytes)<br />
<br />The control of these 3 impacts allows to lengthen the life of the terminals as well as reduce their energy consumption.
<br />The ETSdiff tool allows measuring a differential on these three values and in a given context (database and fixed measurement environment).
<br />The results generated by ETSdiff must help define the interest of the rule reported by Sonarqube in the context of the code analyzed.
<br />
<figure style="text-align: center;">
    <img src="https://live.staticflickr.com/65535/52622379586_f84c767111_o.png" alt="ETSdiff percent comparison" />
    <figcaption style="width: auto;">1 GB database</figcaption>
</figure>