<p>Do not call a function when declaring a for-type loop in order to avoid function calls each iterations. It saves CPU cycles.</p>
<h2>Noncompliant Code Example</h2>
<pre>
for ($i = 0; $i <= foo(); $i++) { // NOK
  ......
}
</pre>
<h2>Compliant Solution</h2>
<pre>
$maxI = foo();
for ($i = 0; $i <= $maxI; $i++) {
  .....
}

  OR

for ($i = 0, $maxI = foo(); $i <= $maxI; $i++) {
  .....
}
}</pre>

<h2>The three sources of impacts of a code identified are: </h2>
- Energy (measured in joules)
<br />    - Transfer (measured in Bytes)
<br />    - Storage (measured in Bytes)<br />
<br />The control of these 3 impacts allows to lengthen the life of the terminals as well as reduce their energy consumption.
<br />The ETSdiff tool allows measuring a differential on these three values and in a given context (database and fixed measurement environment).
<br />The results generated by ETSdiff must help define the interest of the rule reported by Sonarqube in the context of the code analyzed.
<br />

<figure style="text-align: center;">
  <img src="https://live.staticflickr.com/65535/52622634654_bf3c3d9ba8_o.png" alt="ETSdiff percent comparison" />
  <figcaption style="width: auto;">1 GB database</figcaption>
</figure>
