<p>If possible, the utilisation of svg image format (or <svg/> html tag) is recommended over other image format.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public void foo() {
        ...
        img_jpg = "image.jpg"
        ...
    }

</pre>
<h2>Compliant Solution</h2>
<pre>

    public void foo() {
        ...
        img_svg = "image.svg"
        ...
   }

</pre>
