<p>When iterating over any collection, fetch the size of the collection in advance to avoid fetching it on each iteration, this saves CPU cycles, and therefore consumes less power.</p>

<h2>Noncompliant Code Example</h2>
<pre>
	$food = array('orange', 'banana', 'apple', 'carrot', 'collard', 'pea');

	// FOR STATEMENTS
	for ($i = 0; $i < count($food); $i++) {
		var_dump($food[$i]);
	}

	// WHILE STATEMENTS
	$i = 0;
	while($i < count($food))
	{
		var_dump($food[$i]);
		$i++;
	}

	// DO WHILE STATEMENTS
	$i = 0;
	do {
		var_dump($food[$i]);
		$i++;
	} while ($i < count($food));

</pre>

<h2>Compliant Solution</h2>
<pre>
	$food = array('orange', 'banana', 'apple', 'carrot', 'collard', 'pea');
	// FOR STATEMENTS
	$size = sizeof($food);
	for ($i = 0; $i < $size; $i++) {
		var_dump($food[$i]);
	}

	// WHILE STATEMENTS
	$i = 0;
	$size = count($food);
	while($i < $size)
	{
		var_dump($food[$i]);
		$i++;
	}

	// DO WHILE STATEMENTS
	$i = 0;
	$size = count($food);
	do {
		var_dump($food[$i]);
		$i++;
	} while ($i < $size);

</pre>
