<p>Avoid the use of try-catch-finally when not needed</p>
<h2>Noncompliant Code Example</h2>
<pre>
        try {
            int i = 0;
            System.out.println((char) i); //println() doesn't send an Exception
            }
        } catch (IOException e) {
            ....... //never reach
        } finally {
            ...... //never reach
        }
</pre>
<h2>Compliant Solution</h2>
<pre>
        int i = 0;
        System.out.println((char) i);
</pre>
