<p>Do not execute an SQL request in a loop</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public void foo() {
        for (int i = 0; i < 20; i++) {  // Noncompliant
            System.out.println(i);
            boolean b = getMyValue() > 6;
        }
    }

</pre>
<h2>Compliant Solution</h2>
<pre>

    public void foo() {
        int myValue =  getMyValue();
        for (int i = 0; i < myValue; i++) {
            System.out.println(i);
            boolean b = getMyValue() > 6;
        }
    }

</pre>
